lib/algoliasearch-rails.rb

```diff
@@ -561,14 +561,10 @@ def algolia_reindex(batch_size = AlgoliaSearch::IndexSettings::DEFAULT_BATCH_SIZ
         tmp_options.delete(:per_environment) # already included in the temporary index_name
         tmp_settings = settings.dup
 
-        if options[:check_settings] == false
-          ::Algolia::copy_index!(src_index_name, tmp_index_name, %w(settings synonyms rules))
-          tmp_index = SafeIndex.new(tmp_index_name, !!options[:raise_on_failure])
-        else
-          tmp_index = algolia_ensure_init(tmp_options, tmp_settings, master_settings)
-        end
+        ::Algolia::copy_index!(src_index_name, tmp_index_name, %w(settings synonyms rules))
+        tmp_index = SafeIndex.new(tmp_index_name, !!options[:raise_on_failure])
```

Comment:
Did you understand why @julienbourdeau introduced this test in https://github.com/algolia/algoliasearch-rails/commit/ba94561f879071b3aecb50eccea76c162a11f700 ? I'm not sure I get it, even though I think I'm fine with your updated version.

Reply:
My understanding was that it was to avoid having to make multiple calls to `get_settings`, but the problem was that the configuration described in the model, even though it might have changed later, was in that case pulled back and that was what caused the issue I'm trying to fix. But what I'm not not sure I understand indeed was the need to pull back this configuration. I may be changing the behaviour and I do need @julienbourdeau enlighten on this.

Reply:
As far as I understand, the new version is not going to apply setting changes anymore. I added the `check_settings` options to avoid  many get/set settings call. Scoped copy indexed feature was just introduced.

Issues about `searchableAttributes` vs `attributeToIndex` should be addressed by updating the code in the algoliasearch block to use the new name.

I this this PR is going to create more issue that it solves. We should definitely encourage people to turn off the `check_settings` feature. Especially, if the "fix" is to ensure people use the code for "check_settings = false".